server/data/tokenStore/inMemoryTokenStore.ts

```diff
@@ -0,0 +1,17 @@
+import TokenStore from './tokenStore'
+
+export default class InMemoryTokenStore implements TokenStore {
+  map = new Map<string, { token: string; expiry: Date }>()
+
+  public async setToken(key: string, token: string, durationSeconds: number): Promise<void> {
+    this.map.set(key, { token, expiry: new Date(Date.now() + durationSeconds * 1000) })
+    return Promise.resolve()
+  }
+
+  public async getToken(key: string): Promise<string> {
+    if (!this.map.has(key) || this.map.get(key).expiry.getTime() < Date.now()) {
+      return Promise.resolve(null)
```

Comment:
I appreciate this isn't your code and is in the template, but wrapping the return value in the `Promise.resolve()` is unnecessary here since `async` will do that automatically.

Reply:
Yeah; agreed.
I think my previous refactorings were around improving various bits of code (not sure whether I made this change or not in my previous version); but I think what I've seen here is that there are some classes etc that we shouldn't touch too much because it will deviate from the template too much, which in turn makes updating the project from the template hard!
I think perhaps a better approach is that we should look to make improvements in the template project, then let them filter down into the various projects over time. Thats perhaps the better way of doing what I was trying to do in my previous refactorings 😁